content/blog/2018-12-25-darktable-2.6/2018-12-25-darktable-2.6.md

```diff
@@ -0,0 +1,1167 @@
+comparison.
+
+Notice that the thumbnails are only updated when you leave an image,
+so the thumbnail for the image being edited is usually not up to date.
+
+<span id="colorbalance" />
+## Revamp of the *color balance* module
+
+The color balance has been considerably improved. Although its name
+contains "color", it is actually a much more general module. It can
+adjust levels pretty much like the *levels* module, and can now
+also tweak the contrast with an S-shaped curve. Obviously, you can
+also still adjust the colors to add or remove a color cast in the
+shadows, highlight and midtones separately.
+
+The module gains two modes to work in ProPhotoRGB mode. Also, you now
+have the choice between the old RGBL (Red, Green, Blue, Luma) sliders,
+and HSL (Hue, Saturation, Luma).
+
+Let's play with the module on an image with multiple white balances.
+This is the original image, with only the basic modules activated, and
+base curve disabled:
+
+  <div style="text-align: center">
+  ![]({filename}cb-active-modules.jpg)
+  </div>
+
+The snow is obviously white in the real scene, but the snow exposed to
+the sun reflects the sun's light, while the one in the shadow reflects
+the sky's light, much bluer. The color picker module in the right
+sidebar of the darkroom allows visualizing and quantifying these color
+casts:
+
+  <div style="text-align: center">
+  ![]({filename}cb-patches-original.png)
+  </div>
+
+(The negative value for the 'b' channel represents the blue color)
+
+The *color balance* module now has color pickers to neutralize the
+colors. On this image, the global optimizer works rather well. After
+one click on "neutralize colors", the blue cast in the shadows is
+reduced:
+
+  <div style="text-align: center">
+  ![]({filename}cb-picker-neutralize.jpg)
+  </div>
+
+Looking more carefully at the patches selected by the picker, notice
+that the 'b' value is now much closer to 0:
+
+  <div style="text-align: center">
+  ![]({filename}cb-patches-neutralized.png)
+  </div>
+
+If the global optimizer guesses wrong, it is possible to specify color
+patches for highlights, shadows and mid-tones (preferably in this
+order) separately with the corresponding color pickers, and if needed
+re-launch the "neutralize colors" (called "neutralize colors from
+patches" once you have selected these patches).
+
+Similarly, the tones can be adjusted similarly to the *levels* module,
+either with the global "optimize luma" or with individual pickers.
+
+The "master" section at the top allows adjusting the global contrast
+and saturation. "contrast fulcrum" and "contrast" apply an S-shaped
+curve, centered around the fulcrum, and with a slope given by the
+contrast. In other words, with a positive contrast, parts of the image
+below the fulcrum will be darkened and parts above it will have their
+luminance increased:
+
+  <div style="text-align: center">
+  ![]({filename}cb-contrast.jpg)
+  </div>
+
```

Comment:
the real goal of color balance is to perform color-grading, as in movies. See the orange/teal presets and this : http://juanmelara.com.au/blog/the-summer-blockbuster-colour-grading-tutorial